common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/CurrentTaskAuthorizationFragment.java

```diff
@@ -0,0 +1,187 @@
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentKey.REQUEST_CODE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AuthorizationIntentKey.RESULT_CODE;
+
+/**
+ * This base classes
+ * - handles how AuthorizationFragments communicates with the outside world.
+ * - handles basic lifecycle operations.
+ */
+public abstract class CurrentTaskAuthorizationFragment extends Fragment {
+
+    private static final String TAG = AuthorizationFragment.class.getSimpleName();
+
+    /**
+     * The bundle containing values for initializing this fragment.
+     */
+    private Bundle mInstanceState;
+
+    /**
+     * Listens to an operation cancellation event.
+     */
+    private BroadcastReceiver mCancelRequestReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            cancelAuthorization(intent.getBooleanExtra(REQUEST_CANCELLED_BY_USER, false));
+        }
+    };
+
+    void setInstanceState(@NonNull final Bundle instanceStateBundle) {
+        mInstanceState = instanceStateBundle;
+    }
+
+    @SuppressFBWarnings(FindBugsConstants.NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        final String methodName = "#onCreate";
+        super.onCreate(savedInstanceState);
+
+        // Register Broadcast receiver to cancel the auth request
+        // if another incoming request is launched by the app
+        LocalBroadcastManager.getInstance(getContext()).registerReceiver(mCancelRequestReceiver,
+                new IntentFilter(CANCEL_INTERACTIVE_REQUEST));
+
+        if (savedInstanceState == null) {
+            Logger.verbose(TAG + methodName, "Extract state from the intent bundle.");
+            extractState(mInstanceState);
+        } else {
+            // If activity is killed by the os, savedInstance will be the saved bundle.
+            Logger.verbose(TAG + methodName, "Extract state from the saved bundle.");
+            extractState(savedInstanceState);
+        }
+    }
+
+    @SuppressFBWarnings(FindBugsConstants.NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
+    void finish() {
+        final FragmentActivity activity = getActivity();
+        if (activity instanceof AuthorizationActivity) {
+            activity.finish();
+        } else {
+            // The calling activity is not owned by MSAL/Broker.
+            // Just remove this fragment.
+            if (getFragmentManager() != null) {
+                getFragmentManager()
+                        .beginTransaction()
+                        .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
+                        .remove(this)
+                        .commit();
+            }
+        }
+    }
+
+    void extractState(@NonNull final Bundle state) {
```

Comment:
some javadoc here would be nice

Reply:
Sure... I suppose I can change that in the original class as well ;-)